mesosdriver/executor_driver.go

```diff
@@ -83,23 +82,49 @@ func (driver *ExecutorDriver) unacknowledgedUpdates() (result []executor.Call_Up
 
 // eventLoop is the main handler event loop of the driver. Called from Run()
 func (driver *ExecutorDriver) eventLoop(decoder encoding.Decoder,
-	h events.Handler) (err error) {
+	h events.Handler) error {
+
+	var err error
 
 	log.Info("Entering event loop")
-	ctx := context.TODO()
+	ctx := context.Background()
+
+	event := make(chan error)
 
-	for err == nil && !driver.shouldQuit {
-		// housekeeping
-		driver.sendFailedTasks()
+	go func() {
+		for {
+			var e executor.Event
+			if err := decoder.Decode(&e); err == nil {
```

Comment:
This is, stupidly, a blocking call. I don't see a nice way to make it a non-blocking, or cancelable call since it doesn't take a context. So, the solution is to wrap it in a goroutine and make that cancelable at least after the next event arrives. At least it won't block us from exiting the `eventLoop()` any more and we can exit the process before waiting for any more events.